### PR TITLE
Refactor fact

### PR DESCRIPTION
**Pull request by Phil-Friderici**

### CODE REVIEW

**Comment by ghoneycutt** [comment]
Are you seeing the unit test timeout with ruby 2.3.1? I'm seeing that with python.

https://github.com/stankevich/puppet-python/issues/336

You can allow ruby 2.3.1 to fail with this

https://github.com/stankevich/puppet-python/pull/337/files

**Comment by Phil-Friderici** [comment]
The downsize would be that Travis is waiting around 10 minutes before failing. This will slow down the testing results quite a lot :(

**Comment by Phil-Friderici** [comment]
Think I will add only puppet latests to run with ruby 2.3.1. This would only cost 10 minutes.

**Comment by Phil-Friderici** [comment]
**Breaking news**
The new version of the fact will only search for sudo within the given systems path.

Maybe time is good for a 1.0.0 release ?

File: spec/unit/facter/sudo_version_spec.rb [hunk 1]
```diff
@@ -1,57 +1,61 @@
...
+  subject(:fact) { Facter.fact(:sudo_version) }
+
+  before :each do
+    # Ensure we're populating Facter's internal collection with our Fact
+    SudoVersion.add_facts
+  end
 
   with_fixtures = {
     'Debian' => {
-      :sudo_output => 'Sudo version 1.8.3p1\nSudoers policy plugin version 1.8.3p1\nSudoers file grammar version 40\nSudoers I/O plugin version 1.8.3p1\n',
```
Comment (ghoneycutt) [inline spec/unit/facter/sudo_version_spec.rb]: shouldn't the fixture stay the same as the output has not changed

Comment (Phil-Friderici) [inline spec/unit/facter/sudo_version_spec.rb]: Removed it for better alignment. Btw: I have seen likewise output on RedHatish systems as well. 
The regex /^Sudo version +(\S+)$/ should be specific enough to not need to test that it ignores the rest of the string.

Comment (ghoneycutt) [inline spec/unit/facter/sudo_version_spec.rb]: True, but we wouldn't know without the whole output. What if we want to tweak our regex? When it comes to fixtures, I think we should take the entire output.

**Comment by ghoneycutt** [comment]
+1 for a v1.0.0 release

File: spec/unit/facter/sudo_version_spec.rb (hunk 1 above)
Comment (Phil-Friderici) [inline spec/unit/facter/sudo_version_spec.rb]: you'll like my latest change :)

**Comment by Phil-Friderici** [comment]
The fact does proactivly search for /opt/quest/bin/sudo again. This ensures fully backward compatibility again :)

**Comment by ghoneycutt** [comment]
+1

**Comment by Phil-Friderici** [comment]
moving on to https://github.com/Ericsson/puppet-module-sudo/pull/32 to prepare 1.0.0 release

**Pull request closed without merging**